Xamarin.Forms.Core/Picker.cs

```diff
@@ -150,8 +150,10 @@ void OnItemDisplayBindingChanged(BindingBase oldValue, BindingBase newValue)
 
 		void OnItemsCollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
 		{
-			SelectedIndex = SelectedIndex.Clamp(-1, Items.Count - 1);
-			UpdateSelectedItem();
+			// Changing the SelectedIndex property is sent to a queue that is not instantly executed.
+			// In the UpdateSelectedItem method, we need the actual selectedIndex.
+			var selectedIndex = SelectedIndex = SelectedIndex.Clamp(-1, Items.Count - 1);
+			UpdateSelectedItem(selectedIndex);
```

Comment:
so you only have to invoke `UpdateSelectedIndex()` when the source the index hasn't been changed